packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -49,18 +48,11 @@ export default function createExPlatClientReactHelpers(
 			return () => {
 				isSubscribed = false;
 			};
-			// We don't expect experimentName to ever change and if it does we want to assignment to stay constant.
+			// experimentName can change for experiments with dates in their name, such as evergreen A/As
+			// but we don't want the assignment to change when it flips over, so we purposely don't include experimentName as a dep here.
 			// eslint-disable-next-line react-hooks/exhaustive-deps
 		}, [] );
 
-		if ( experimentName !== previousExperimentName ) {
```

Comment:
Maybe we should keep the check if both names don't start with `explat_test_aa_weekly_calypso_`? Dynamically created experiment names should be pretty rare.

Reply:
I am going to add a general exception for `explat_test_`